optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -111,8 +124,118 @@ def _from_pretrained(
         local_files_only: bool = False,
         subfolder: str = "",
         trust_remote_code: bool = False,
+        *model_args,
         **kwargs,
     ):
+        device_map = kwargs.get("device_map", "cpu")
+        use_xpu = True if device_map == torch.device("xpu") or device_map == "xpu" else False
+
+        quantization_config = kwargs.pop("quantization_config", None)
+        if not isinstance(config, PretrainedConfig):
+            config, _ = AutoConfig.from_pretrained(
+                model_id,
+                return_unused_kwargs=True,
+                **kwargs,
+            )
+        if hasattr(config, "quantization_config"):
+            if config.quantization_config is None:
+                logger.warning(
+                    "Quantization_config loading failed. If you want to load saved "
+                    "low bit model, please check your quantizate_config.json."
+                )
+            else:
+                logger.info("quantization_config: {}".format(config.quantization_config))
+
+                try:
+                    from neural_compressor.transformers.models.modeling_auto import (
+                        _BaseINCAutoModelClass,
+                    )
+
+                    _BaseINCAutoModelClass.ORIG_MODEL = cls.auto_model_class
+                    model = _BaseINCAutoModelClass.load_low_bit(
+                        model_id,
+                        subfolder=subfolder,
+                        revision=revision,
+                        cache_dir=cache_dir,
+                        token=token,
+                        local_files_only=local_files_only,
+                        force_download=force_download,
+                        trust_remote_code=trust_remote_code,
+                        config=config,
+                        **kwargs,
+                    )
+                    logger.info("Saved low bit model loading successfully. Other input args " "will be ignored.")
+                    return model
+                except Exception as e:
+                    logger.error(e)
+                    logger.error("Saved low bit model loading failed, please check your model.")
+                    exit(0)
+        if isinstance(quantization_config, (RtnConfig, GPTQConfig)):
```

Comment:
I think would be nice to have a function for WOQ and one for static quantization if possible to make `_from_pretrained` a bit less bloated

Reply:
I agree, move it to quantization.py and apply other changes.